File: Programmers/Implement/FailureRate.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <utility>
using namespace std;

bool comp(pair<float, int> a, pair<float, int> b) {
	if (a.first > b.first)
		return true;

	else if (a.first == b.first)
		return (a.second < b.second);

	else
		return false;
}

vector<int> solution(int N, vector<int> stages) {
	vector<int> answer;

	int done = stages.size();
	int yet[505] = {};

	for (int i = 0; i < stages.size(); i++) {
		yet[stages[i]]++;
	}

	vector<pair<float, int> > v;
	for (int i = 1; i <= N; i++) {

		if (yet[i] == 0) {
			v.push_back(make_pair(0, i));
		}
		else {
			float var = (float)yet[i] / done;
			v.push_back(make_pair(var, i));
			done -= yet[i];
		}
	}

	sort(v.begin(), v.end(), comp);
	for (int i = 0; i < v.size(); i++) {
		answer.push_back(v[i].second);
	}
	return answer;
}

File: Programmers/Implement/FailureRate.js
function solution(N, stages) {
    let answer = [];
    const arr = stages.sort(),
        atStage = new Array(N).fill(0),
        fail = [];
    
    arr.forEach(v => {
        atStage[v-1]++;
    });
    
    let len = stages.length;
    let people= 0;
    
    for(let i=0; i<N; i++) {
        if(len-people===0) {
            continue;
        }
        fail.push({
            idx: i+1,
            val: atStage[i]/(len-people)
        });
        people += atStage[i];
    }
    
    answer = fail.sort((a,b) => {
        if(a.val > b.val) {
            return -1;    
        } else if(a.val===b.val) {
            if(a.idx < b.idx) {
                return -1;
            }
            return 1;
        }
        return 1;
    }).map(v=> v.idx);
    
    let fl = fail.length;
    while(fl!==N) {
        answer.push(++fl);
    }
    
    return answer;
}
